cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -247,39 +252,26 @@ public String getHeadlessServiceName() {
      * @return The selector labels as an instance of the Labels object.
      */
     public Labels getSelectorLabels() {
-        return labels.withName(name).strimziSelectorLabels();
+        return getLabelsWithStrimziName(name, Collections.emptyMap()).strimziSelectorLabels();
     }
 
-    /**
-     * @return The selector labels as Map.
-     */
-    public Map<String, String> getSelectorLabelsAsMap() {
-        return getSelectorLabels().toMap();
+    protected Labels getLabelsWithStrimziName(String name, Map<String, String> additionalLabels) {
+        return labels.withStrimziName(name).withAdditionalLabels(additionalLabels);
     }
 
-    protected Map<String, String> getLabelsWithName() {
-        return getLabelsWithName(name);
+    protected Labels getLabelsWithNameAndDiscovery(String name, Map<String, String> additionalLabels) {
```

Comment:
I think Javadoc to explain what _discovery_ is in this context would be helpful.